### DISCUSSION TITLE
Cannot read property 'dispatch' of undefined

### DISCUSSION THREAD

**Discussion by Chris-Jcc** [react-navigation/react-navigation#9438] [Q&A]
First at all, I'm sorry for my english, my english is not good. So here is my problem:
Used in RN code AppRegistry.registerComponent After registering a component, the native side jumps to this page. However, after entering the RN page, there is no navigation in props. The jump code written by my colleagues is as follows. How can I solve this problem? Tks!

jump page function:
let _navigator;

function setTopLevelNavigator(navigatorRef) {
  _navigator = navigatorRef;
}

function navigate(routeName, params) {
  _navigator.dispatch(
    NavigationActions.navigate({
      routeName,
      params,
    }),
  );
}

In page:
navigate('xxx')

**Comment by raajnadar** [comment]
The value of navigation ref is not defined! It will be helpful if you provide a snack repro.
